src/firefly/js/visualize/region/Region.js

```diff
@@ -0,0 +1,291 @@
+ *                                            or 2 radii for ellipse and ellipse annulus
+ * @param angle RegionValue
+ * @param type  RegionType
+ * @param options   RegionOptions, properties of region
+ * @param highlighted bool
+ * @constructor
+ */
+export var Region =
+    ( {wpAry = null,
+      radiusAry = null,
+      dimensionAry = null,
+      angle = null,
+      type = RegionType.undefined,
+      options = null,
+      highlighted = false}
+    ) => ({type, wpAry, radiusAry, dimensionAry, angle, options, highlighted});
+
+/**
+ *
+ * @param color     string, 'white', 'black', 'red', 'green', 'blue', 'cyan', 'magenta', 'yellow'
+ * @param text      string
+ * @param font      RegionFont
+ * @param pointType     RegionPointType
+ * @param pointSize     number
+ * @param editable      bool
+ * @param movable       bool
+ * @param rotatable     bool
+ * @param highlightable bool
+ * @param deletable     bool
+ * @param fixedSize     bool
+ * @param include       bool
+ * @param lineWidth     number
+ * @param offsetX       number
+ * @param offsetY       number
+ * @param message       string, containing parsing message
+ * @constructor
+ */
+
+export var RegionOptions =
+    ( {color = 'blue',
+        text  = null,
+        font  = null,
+        pointType = null,
+        pointSize =  0,
+        editable  =  true,
+        movable   =  true,
+        rotatable =  true,
+        highlightable =  true,
+        deletable = true,
+        fixedSize = false,
+        include   = true,
+        lineWidth = 0,
+        offsetX   = 0,
+        offsetY   = 0,
+        message  = null}
+    ) => ({color, text, font, pointType, pointSize, editable, movable, rotatable, highlightable,
+        deletable, fixedSize, include, lineWidth, offsetX, offsetY, message});
+
+/**
+ *
+ * @param value value for radius and dimension
+ * @param unit  DEGREE, SCREEN_PIXEL and IMAGE_PIXEL
+ * @constructor
+ */
+
+export var RegionValue =
+    (value = 0.0, unit = RegionValueUnit) => ({value, unit});
+
+/**
+ *
+ * @param width  RegionValue width of box or first radius of ellipse
+ * @param height RegionValue height of box or second radius of ellipse
+ * @constructor
+ */
+export var RegionDimension = (width, height) => ({width, height});
+
+/**
+ *
+ * @param name  string
+ * @param point  string for font size
+ * @param weight string
+ * @param slant  string
+ * @constructor
+ */
+export var RegionFont = ({name = 'SansSerif', point = '10', weight = 'bold', slant = 'italic'}) =>
+                        ( {name, point, weight, slant} );
+
+
+/**
+ * regiopn parse exception
+ */
+export class RegParseException extends Error {
+    constructor(message) {
+        super(message);
+        this.message = message;
+        this.name = 'RegionParseError';
+    }
+}
+
+
+var findEnumKey = (enumObj, val) => ( enumObj.enums.find((item) => item.key.toLowerCase() === val.toLowerCase()));
```

Comment:
You don't need this.  
```
 RegionType = new Enum([ 'circle', 'annulus', 'ellipse', 'ellipseannulus', 'box',
                                    'boxannulus', 'line', 'point', 'polygon',
                                     'text', 'undefined'], 
                      { ignoreCase: true });
```
then
```
RegionType.get('AnnLus'); // => RegionType.annulus
```
see https://www.npmjs.com/package/enum
about half way down.